socketio/src/client/client.rs

```diff
@@ -468,6 +568,53 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn socket_io_reconnect_integration() -> Result<()> {
+        let url = crate::test::socket_io_restart_server();
+
+        let socket = ClientBuilder::new(url).connect();
+        assert!(socket.is_ok(), "should connect success");
+        let socket = socket.unwrap();
+
+        std::thread::sleep(std::time::Duration::from_millis(200));
+
+        let r = socket.emit("message", json!(""));
+        assert!(r.is_ok(), "should emit message success");
+
+        let r = socket.emit("restart_server", json!(""));
+        assert!(r.is_ok(), "should emit restart success");
+
+        let mut is_emit_error = false;
+        for _ in 0..10 {
+            std::thread::sleep(std::time::Duration::from_millis(500));
+            let r = socket.emit("message", json!(""));
+            if r.is_err() {
+                is_emit_error = true; // server is down
+                break;
+            }
+        }
+
+        assert_eq!(
+            is_emit_error, true,
+            "should emit failed because server is down"
+        );
+
+        for _ in 0..10 {
```

Comment:
suggestion: why is this looped? Should the next message right after the failure return success?

Reply:
@nshaaban-cPacket Server not start immediately, emit will not success until server is start. If I change the test server to restart immediately, user code may emit success every time, we have no idea if the server is restart or not.

Reply:
(usability issue) seems difficult for users of the library to use